torchtext/vocab.py

```diff
@@ -216,6 +218,33 @@ def set_vectors(self, stoi, vectors, dim, unk_init=torch.Tensor.zero_):
                 self.vectors[i] = unk_init(self.vectors[i])
 
 
+def build_vocab(dataset, field, data_name, **kwargs):
+    """Construct the Vocab object for the field from a dataset.
+
+    Arguments:
+        dataset: Dataset with the iterable data.
+        field: Field object with the information of the special tokens.
+        data_name: The name of data used to build vocab (e.g. 'text', 'label').
+        Remaining keyword arguments: Passed to the constructor of Vocab.
+
+    Examples:
+        >>> field.vocab = build_vocab(dataset, field, 'text')
+    """
+    counter = Counter()
+    for x in dataset:
+        x = getattr(x, data_name)
+        if not field.sequential:
+            x = [x]
+        try:
+            counter.update(x)
+        except TypeError:
+            counter.update(chain.from_iterable(x.text))
+    specials = list(OrderedDict.fromkeys(
+        tok for tok in [field.unk_token, field.pad_token, field.init_token,
+                        field.eos_token] if tok is not None))
+    return Vocab(counter, specials=specials, **kwargs)
```

Comment:
Vocab does a lot of things that seem unrelated to dictionaries, for example "load_vectors" and "set_vectors", "__eq__" etc.  - maybe we should audit it and see if we can remove a few things for 0.4?

Vocab is also undocumented and not referenced by anything, even though it seems pretty useful.

Reply:
Let me see if I can make some changes there. I cannot estimate the efforts to split vacab and vector etc.

Reply:
It's ok if we build a very minimal and advertised Dictionary instead of replicating all features of Vocab. That will also give us the chance to make it fast.